alerting/src/main/kotlin/org/opensearch/alerting/TriggersActionThresholdUtils.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.alerting.util
+
+object TriggersActionThresholdUtils {
+
+    /**
+     Get threshold value
+     */
+    fun getThreshold(
+        p: TriggersActionThresholdParams,
+        currentTriggerActionSize: Int,
+        triggerMaxActions: Int,
+        triggerTotalMaxActions: Int
+    ): Int {
+        var threshold = currentTriggerActionSize
+        if (triggerTotalMaxActions == 0) {
+            threshold = 0
+        } else if (triggerTotalMaxActions > 0) {
+            threshold = if (triggerMaxActions >= 0) {
+                val temporaryThreshold = if (triggerMaxActions < p.surplusActionCount) triggerMaxActions else p.surplusActionCount
+                if (temporaryThreshold < currentTriggerActionSize) temporaryThreshold else currentTriggerActionSize
```

Comment:
Use kotlin [Math.min](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.math/min.html) function to simplify this,